### ISSUE TITLE
Add support for ncRNAs in the TranscriptDB generation script

### ISSUE THREAD

**Issue by khughitt** [elsayed-lab/eupathdb-organismdb#1]
Currently, the ~~`makeTxDbPackage`~~ `makeTranscriptDbFromGFF` used to generate TxDb packages only includes entries corresponding to `mRNA` entries in the input GFF.

The result is that ncRNA GFF entries (which may only have exon, gene, and transcript rows) will be excluded from the database.

In other TxDb databases, however, ncRNAs are usually included, e.g:

```r
library(Homo.sapiens)
head(select(Homo.sapiens, keys=keys(Homo.sapiens, keytype='GENEID'), columns=c('GENENAME', 'TXSTART'), keytype='GENEID'), 30)
```

Example protein-coding entry from [TriTrypDB-9.0_LmajorFriedlin.gff](http://tritrypdb.org/common/downloads/release-9.0/LmajorFriedlin/gff/data/TriTrypDB-9.0_LmajorFriedlin.gff):

```
LmjF.33	TriTrypDB	mRNA	1379369	1384630	.	+	.	ID=rna_LmjF.33.2910-1;Name=LmjF.33.2910-1;description=LmjF.33.2910-1;size=5262;Parent=LmjF.33.2910;Dbxref=ApiDB:LmjF.33.2910,NCBI_gi:157875315,NCBI_gi:68129127,taxon:347515
LmjF.33	TriTrypDB	CDS	1379369	1384630	.	+	0	ID=cds_LmjF.33.2910-1;Name=cds;description=.;size=5262;Parent=rna_LmjF.33.2910-1
LmjF.33	TriTrypDB	exon	1379369	1384630	.	+	.	ID=exon_LmjF.33.2910-1;Name=exon;description=exon;size=5262;Parent=rna_LmjF.33.2910-1
LmjF.33	TriTrypDB	gene	1390098	1391033	.	+	.	ID=LmjF.33.2920;Name=LmjF.33.2920;description=hypothetical+protein%2C+conserved;size=936;web_id=LmjF.33.2920;locus_tag=LmjF.33.2920;size=936;Alias=LmjF33.2920:pep,157875317,LmjF33.2920:mRNA,68129128,LmjF33.2920,LmjF.33.2920
```

Example ncRNA entry:

```
LmjF.33	TriTrypDB	gene	95481	97247	.	-	.	ID=LmjF.33.ncRNA1;Name=LmjF.33.ncRNA1;description=unspecified+product;size=1767;web_id=LmjF.33.ncRNA1;locus_tag=LmjF.33.ncRNA1;size=1767;Alias=LmjF.33.ncRNA1,Lmjchr33_ncRNA1,Lmjchr33_ncRNA1:ncRNA
LmjF.33	TriTrypDB	transcript	95481	97247	.	-	.	ID=rna_LmjF.33.ncRNA1-1;Name=LmjF.33.ncRNA1-1;description=LmjF.33.ncRNA1-1;size=1767;Parent=LmjF.33.ncRNA1;Dbxref=ApiDB:LmjF.33.ncRNA1,taxon:347515
LmjF.33	TriTrypDB	exon	95481	97247	.	-	.	ID=exon_LmjF.33.ncRNA1-1;Name=exon;description=exon;size=1767;Parent=rna_LmjF.33.ncRNA1-1
```

**Comment by khughitt** [comment]
@mrjc42 Any suggestions for working around this?

It seems like I could probably set `useGenesAsTranscripts=TRUE`, but this will only work for some species, as suggested in the docs.

**Comment by mrjc42** [comment]
Hi Keith,

So Herve recently pushed some changes to the devel branch (major changes) that cleans up and speeds up the code for processing GFFs into TranscriptDb objects.   Among the many changes are changes that allow for a more inclusive usage of a GFFs available data like the data you mentioned here. 

In the future, please post questions like this either to our support site here:

https://support.bioconductor.org/

Or to our dev mailing list (which you can subscribe to here):

http://www.bioconductor.org/help/support/#bioc-devel

Thanks for your interest,

 Marc

**Comment by khughitt** [comment]
Hi Marc,

Thanks for the heads up about the devel branch. I was planning to look into that and possibly submit some patches to generalize the package helper functions, but it's nice to see someone has beat me to it!

I wasn't sure how useful this question would be to the general public so I decided to post here instead of https://support.bioconductor.org/. Next time I will go ahead and post straight there though.

Cheers,
Keith

**Comment by mrjc42** [comment]
That is a good plan.  It turns out that there are a LOT of people using bioconductor.  So for many questions the odds are pretty good that if you are wondering about something that someone else might be wondering that too.  This is why we ask people to use the support site.  We want other people who may have similar questions to also benefit from our answers.

  Marc

**Comment by khughitt** [comment]
Fixed in 54b914f0d1c31fcf389cf491d199f4c427264d24
